src/actions/shelf.ts

```diff
@@ -46,21 +47,36 @@ export type ShelfFieldMove = ReduxAction<typeof SHELF_FIELD_MOVE, {
  * Change Function of a FieldDef to a specific value.
  */
 export const SHELF_FUNCTION_CHANGE = 'SHELF_FUNCTION_CHANGE';
-
 export type ShelfFunctionChange = ReduxAction<typeof SHELF_FUNCTION_CHANGE, {
   shelfId: ShelfId,
   fn: ShelfFunction;
 }>;
 
-export const SHELF_SPEC_LOAD = 'SHELF_SPEC_LOAD';
-export type ShelfSpecLoad = ReduxAction<typeof SHELF_SPEC_LOAD, {
-  spec: FacetedCompositeUnitSpec
+export const SHELF_FUNCTION_ADD_WILDCARD = 'SHELF_FUNCTION_ADD_WILDCARD';
+export type ShelfFunctionAddWildcard = ReduxAction<typeof SHELF_FUNCTION_ADD_WILDCARD, {
+  shelfId: ShelfId,
+  fn: ShelfFunction;
+}>;
+
+export const SHELF_FUNCTION_ENABLE_WILDCARD = 'SHELF_FUNCTION_ENABLE_WILDCARD';
+export type ShelfFunctionEnableWildcard = ReduxAction<typeof SHELF_FUNCTION_ENABLE_WILDCARD, {
+  shelfId: ShelfId,
+  fn: ShelfFunction;
```

Comment:
Since SHELF_FUNCTION_DISABLE_WILDCARD does not require `fn`, SHELF_FUNCTION_ENABLE_WILDCARD should not require `fn`.